elytron/src/main/java/org/wildfly/extension/elytron/ElytronSubsystemTransformers.java

```diff
@@ -581,6 +591,24 @@ protected void convertAttribute(PathAddress address, String attributeName, Model
 
     };
 
+    /* [WFCORE-6244] converts list of objects:              [{"from"=STRING, "to"=[STRING,STRING,...]}...]
+     * back to a direct string to string-list mapping:      {STRING=[STRING,STRING,...],...}
+     *
+     * XML schema remains unchanged. */
+    private static final AttributeConverter MAPPED_ROLE_MAPPER_CONVERTER = new AttributeConverter.DefaultAttributeConverter() {
+        @Override
+        protected void convertAttribute(PathAddress address, String attributeName, ModelNode attributeValue, TransformationContext context) {
+            if (attributeValue.isDefined()) {
+                ModelNode oldRoleMappingMap = new ModelNode();
+                for (ModelNode mapping : attributeValue.asList()) {
+                    oldRoleMappingMap.add(mapping.get(FROM).asString(), mapping.get(TO));
```

Comment:
I guess you can just add an additional check to the if sentence so `if (attribute is defined && is List) then ...`
The fact that the attribute is defined is not enough in this case since both types will go throughout the conversion.